src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -2,85 +2,115 @@
 
+            _name = name;
+
+            Modifiers = DeclarationModifiers.Private |
+                (isReadOnly ? DeclarationModifiers.ReadOnly : DeclarationModifiers.None) |
+                (isStatic ? DeclarationModifiers.Static : DeclarationModifiers.None);
+
             _property = property;
-            _hasInitializer = hasInitializer;
+            HasInitializer = hasInitializer;
         }
 
-        public bool HasInitializer
-        {
-            get { return _hasInitializer; }
-        }
+        protected override IAttributeTargetSymbol AttributeOwner
+            => _property;
+
+        internal override Location ErrorLocation
+            => _property.Location;
+
+        protected override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList
+            => _property.CSharpSyntaxNode.AttributeLists;
 
         public override Symbol AssociatedSymbol
-        {
-            get
-            {
-                return _property;
-            }
-        }
+            => _property;
 
         public override ImmutableArray<Location> Locations
-        {
-            get
-            {
-                return _property.Locations;
-            }
-        }
-
-        internal override bool SuppressDynamicAttribute
-        {
-            get
-            {
-                return false;
-            }
-        }
+            => _property.Locations;
 
         internal override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBeingBound)
-        {
-            return _property.Type;
-        }
+            => _property.Type;
 
         internal override bool HasPointerType
+            => _property.HasPointerType;
+
+        internal sealed override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
         {
-            get
+            Debug.Assert((object)arguments.AttributeSyntaxOpt != null);
+
+            var attribute = arguments.Attribute;
+            Debug.Assert(!attribute.HasErrors);
+            Debug.Assert(arguments.SymbolPart == AttributeLocation.None);
+
+            if (attribute.IsTargetAttribute(this, AttributeDescription.FixedBufferAttribute))
             {
-                return _property.HasPointerType;
+                // error CS8362: Do not use 'System.Runtime.CompilerServices.FixedBuffer' attribute on property
+                arguments.Diagnostics.Add(ErrorCode.ERR_DoNotUseFixedBufferAttrOnProperty, arguments.AttributeSyntaxOpt.Name.Location);
             }
+
+            base.DecodeWellKnownAttribute(ref arguments);
         }
 
         internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
         {
             base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
 
+            SynthesizedFieldSymbolBase.AddSynthesizedFieldAttributes(ref attributes, this, suppressDynamicAttribute: false);
```

Comment:
>SynthesizedFieldSymbolBase.AddSynthesizedFieldAttributes(ref attributes, this, suppressDynamicAttribute: false); [](start = 12, length = 112)

It feels like, if we move SourceFieldSymbol.AddSynthesizedAttributes to the new base, we wouldn't have to add this call.